datadotworld/client/api.py

```diff
@@ -441,11 +525,290 @@ def download_datapackage(self, dataset_key, dest_dir):
+        -------
+        NoneType
+            None
+
+        Raises
+        ------
+        RestApiException
+            If a server error occurs
+
+        Examples
+        --------
+        >>> import datadotworld as dw
+        >>> api_client = dw.api_client()
+        >>> api_client.sparql_post('username/test-dataset', query)
+        None
+        """
+        owner_id, dataset_id = parse_dataset_key(dataset_key)
+        try:
+            return self._sparql_api.sparql_post(owner_id, dataset_id, query)
+        except _swagger.rest.ApiException as e:
+            raise RestApiError(cause=e)
+
+    # Download Operations
+
+    def download_dataset(self, dataset_key):
+        """Return a .zip containing all files within the dataset as uploaded.
+
+        Parameters
+        ----------
+        dataset_key : str
+            Dataset identifier, in the form of owner/id
+
+        Returns
+        -------
+        NoneType
+            None
+
+        Raises
+        ------
+        RestApiException
+            If a server error occurs
+
+        Examples
+        --------
+        >>> import datadotworld as dw
+        >>> api_client = dw.api_client()
+        >>> api_client.download_dataset('username/test-dataset')
+        None
+        """
+        owner_id, dataset_id = parse_dataset_key(dataset_key)
+        try:
+            return self._download_api.download_dataset(owner_id, dataset_id)
+        except _swagger.rest.ApiException as e:
+            raise RestApiError(cause=e)
+
+    def download_file(self, dataset_key, file):
+        """Return a file within the dataset as uploaded.
+
+        Parameters
+        ----------
+        dataset_key : str
+            Dataset identifier, in the form of owner/id
+
+        file : str
+            File path to be returned
+
+        Returns
+        -------
+        NoneType
+            None
+
+        Raises
+        ------
+        RestApiException
+            If a server error occurs
+
+        Examples
+        --------
+        >>> import datadotworld as dw
+        >>> api_client = dw.api_client()
+        >>> api_client.download_file('username/test-dataset', '/my/local/example.csv')
+        """
+        owner_id, dataset_id = parse_dataset_key(dataset_key)
+        try:
+            return self._download_api.download_file(owner_id, dataset_id, file)
+        except _swagger.rest.ApiException as e:
+            raise RestApiError(cause=e)
+
     @staticmethod
     def __build_dataset_obj(dataset_constructor, file_constructor, args):
-        files = ([file_constructor(name, url)
-                  for name, url in args['files'].items()]
-                 if 'files' in args else None)
+        labels = args.get('labels')
+        description = args.get('description')
+        if 'files' in args:
+            name = args['files'].keys()[0]
+            url = args['files'].values()[0]
+            files = [file_constructor(name, url, labels=labels, description=description)]
```

Comment:
It does not make sense that the same `description` and set of `labels` would be shared by all files.